src/models/CommonModel.ts

```diff
@@ -234,20 +234,17 @@ export class CommonModel extends CommonSchema<CommonModel> {
   private static mergeItems(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema, alreadyIteratedModels: Map<CommonModel, CommonModel> = new Map()) {
     const merge = (models: CommonModel | CommonModel[] | undefined): CommonModel | undefined => {
       if (Array.isArray(models)) {
-        if (models.length > 0) {
-          let mergedItemsModel: CommonModel | undefined = undefined;
-          models.forEach((model, index) => { 
-            Logger.warn(`Found duplicate items at index ${index} for model. Model item for ${mergeFrom.$id || 'unknown'} merged into ${mergeTo.$id || 'unknown'}`, mergeTo, mergeFrom, originalSchema);
-            mergedItemsModel = CommonModel.mergeCommonModels(mergedItemsModel, model, originalSchema, alreadyIteratedModels); 
-          });
-          return mergedItemsModel;
-        } 
-        return undefined;
+        let mergedItemsModel: CommonModel | undefined = undefined;
+        models.forEach((model, index) => { 
+          Logger.warn(`Found duplicate items at index ${index} for model. Model item for ${mergeFrom.$id || 'unknown'} merged into ${mergeTo.$id || 'unknown'}`, mergeTo, mergeFrom, originalSchema);
+          mergedItemsModel = CommonModel.mergeCommonModels(mergedItemsModel, model, originalSchema, alreadyIteratedModels); 
```

Comment:
Why are you iterating over whole array but using only the last result.? It is unclear for me. Please make it more clear.

Reply:
I am not using the last result, I ensure that all tuple schemas in `items` are merged into one. At least until we start to support tuples: https://github.com/asyncapi/generator-model-sdk/issues/67

Reply:
I think that this should be changed. this mergeCommonModels should not hide mutation of mergedItemsModel.

Reply:
Agreed, should be changed as of that issue. But since this is a refactoring of the simplifier, this functionality is already in master, so let's not modify it here.